src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/ReflectionUtils/TypeExtensions.cs

```diff
@@ -44,6 +52,10 @@ public static bool IsIEnumerable(this Type type)
 
         public static bool IsIDictionary(this Type type)
         {
+            if (type.IsGenericType && type.GetGenericTypeDefinition().Equals(typeof(IDictionary<,>)))
```

Comment:
This method should be a simple check for `Dictionary<,>` since the implementation is not equipped to handle other `IDictionary<,>` types like `SortedDictionary<,>` etc.

Same comments for `IsIList` above